### PR TITLE
[Modules] Updated KeyVault vaults to new dependencies approach

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated KeyVault vaults to new dependencies approach

| Pipeline |
| - |
| [KeyVault: Vaults](https://github.com/Azure/ResourceModules/actions/workflows/ms.keyvault.vaults.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.KeyVault/vaults/.test/common/deploy.test.bicep [hunk 1]
```diff
@@ -0,0 +1,161 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
```
Comment (eriqua) [inline modules/Microsoft.KeyVault/vaults/.test/common/deploy.test.bicep:5]: Should we have a new empty line after each section? Currently no empty line after `Parameters`, one empty line after `Test execution`.

File: modules/Microsoft.KeyVault/vaults/.test/common/dependencies.bicep [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    }
+}
+
+@description('The resource ID of the created Virtual Network Subnet.')
+output subnetResourceId string = virtualNetwork.properties.subnets[0].id
+
+@description('The principal ID of the created Managed Identity.')
+output managedIdentityPrincipalId string = managedIdentity.properties.principalId
+
+@description('The resource ID of the created Virtual Network Subnet.')
```
Comment (eriqua) [inline modules/Microsoft.KeyVault/vaults/.test/common/dependencies.bicep:62]:
```suggestion
@description('The resource ID of the created Private DNS Zone.')
```

File: modules/Microsoft.KeyVault/vaults/.test/common/deploy.test.bicep [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+param resourceGroupName string = 'ms.keyvault.vaults-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints')
+param serviceShort string = 'kvvcom'
+
+// =========== //
+// Deployments //
```
[the remark this answers is not in the document]
Comment (MrMCake) [reply modules/Microsoft.KeyVault/vaults/.test/common/deploy.test.bicep:17]: We could use that too - and agree on a naming with the team. 
Either way I'd suggest to change it once the PRs are in as a bulk edit, and not for the individual one.

File: modules/Microsoft.KeyVault/vaults/.test/common/deploy.test.bicep (hunk 1 above)
Comment (MrMCake) [reply modules/Microsoft.KeyVault/vaults/.test/common/deploy.test.bicep:5]: We could - though I got to ask if this empty line vs. no empty (aka minor formatting) is worth blocking PRs.

File: modules/Microsoft.KeyVault/vaults/.test/common/deploy.test.bicep (hunk 3 above)
[the remark this answers is not in the document]
Comment (eriqua) [reply modules/Microsoft.KeyVault/vaults/.test/common/deploy.test.bicep:17]: I agree, this also applies to the comment below. I'll create an issue for that to keep track.

File: modules/Microsoft.KeyVault/vaults/.test/common/deploy.test.bicep (hunk 1 above)
Comment (eriqua) [reply modules/Microsoft.KeyVault/vaults/.test/common/deploy.test.bicep:5]: As per other comment above, I agree this shouldn't block the PRs, but wanted to at least mention it and keep track of it. The whole formatting and naming can be discussed separately and updated once all PRs, or the majority of them, will be in.

File: modules/Microsoft.KeyVault/vaults/.test/common/deploy.test.bicep (hunk 3 above)
[the remark this answers is not in the document]
Comment (eriqua) [reply modules/Microsoft.KeyVault/vaults/.test/common/deploy.test.bicep:17]: Ref #2038

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**